tests/functional/tests.py

```diff
@@ -169,28 +212,21 @@ def test_list_buckets(client):
                 continue
             raise ValueError('list_bucket api failure')
     except Exception as err:
-            log_output['message'] = err
-            log_output['error'] = traceback.format_exc()
-            log_output['status'] = FAIL
-            log_output['duration'] = duration = round(time.time() - start_time, 2)
-            print_json(log_output)
+            print(log_output.json_report_failure(err))
             exit()
     finally:
         client.remove_bucket(bucket_name)
     # Test passes
-    log_output['duration'] = duration = round(time.time() - start_time, 2)
-    print_json(log_output)
+    print(log_output.json_report_success())
 
 def test_fput_object_small_file(client, testfile):
-    start_time = time.time()
-
-    log_output = new_log_result(client.fput_object)
-    log_output['description'] = 'Tests fput_object api'
+    log_output = LogOutput(client.fput_object)
+    log_output.description = 'Tests fput_object api'
     # Get a unique bucket_name and object_name
-    log_output['args']['bucket_name'] = bucket_name = uuid.uuid4().__str__()
-    log_output['args']['object_name'] = object_name = uuid.uuid4().__str__()
-    log_output['args']['file_path'] = testfile
-    log_output['args']['metadata'] = metadata = {'x-amz-storage-class': 'STANDARD_IA'}
+    log_output.args['bucket_name'] = bucket_name = uuid.uuid4().__str__()
+    log_output.args['object_name'] = object_name = uuid.uuid4().__str__()
+    log_output.args['file_path'] = testfile
+    log_output.args['metadata'] = metadata = {'x-amz-storage-class': 'STANDARD_IA'}
```

Comment:
rather than setting the args one by one for log_output, it would be better to provide a helper method that takes all needed params in one line. example - log_output.set_params(a,b,c). or better still pass all needed parameters in your log_output constructor call. Makes it more readable and maintainable.